packages/fast-components/src/dialog/dialog.ts

```diff
@@ -0,0 +1,178 @@
+import { attr, FastElement, DOM } from "@microsoft/fast-element";
+import { keyCodeEscape, keyCodeTab } from "@microsoft/fast-web-utilities";
+import tabbable from "tabbable";
+
+export class Dialog extends FastElement {
+    @attr
+    public modal: boolean = true;
+
+    @attr
+    public hidden: boolean = false;
+
+    @attr({ attribute: "trap-focus" })
+    public trapFocus: boolean = true;
+
+    @attr({ attribute: "aria-describedby" })
+    public ariaDescribedby: string;
+
+    @attr({ attribute: "aria-labelledby" })
+    public ariaLabelledby: string;
+
+    @attr({ attribute: "aria-label" })
+    public ariaLabel: string;
+
+    public dialog: HTMLDivElement;
+
+    private tabbableElements: HTMLElement[];
+
+    private observer: MutationObserver;
+
+    public dismiss(): void {
+        this.dispatchEvent(
+            new CustomEvent("dismiss", {
+                bubbles: true,
+                composed: true,
+            })
+        );
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+
+        // store references to tabbable elements
+        this.tabbableElements = tabbable(this);
+
+        this.observer = new MutationObserver(this.onChildListChange);
+        // only observe if nodes are added or removed
+        this.observer.observe(this, { childList: true });
+
+        document.addEventListener("keydown", this.handleDocumentKeydown);
+
+        // Ensure the DOM is updated
+        // This helps avoid a delay with `autofocus` elements recieving focus
+        DOM.queueUpdate(this.trapFocusChanged);
```

Comment:
@EisenbergEffect just an FYI that DOM.queueUpdate solved the issue seen with autofocus elements being delayed. Thx!

Reply:
Groovy. I'm coming up on reviewing this code soon. I'll take a look at the internal details and make sure my guess on why this is, is correct and see if we need to make any changes in the library or not.